docs/manual/index-checker.tex

```diff
@@ -338,6 +338,26 @@ \section{Upper bounds\label{index-upperbound}}
   is less than the length of the given sequences with the given offsets.
 \end{description}
 
+The Upper Bound Checker also supports a declaration annotation
+\refqualclasswithparams{checker/index/qual}{HasSubsequence}{String[] value, String[] from, String[] to}. This annotation
+can be written on sequence declarations to indicate that a sequence has a named subsequence with the given name,
+which ranges from \code{from} to \code{to}. So, for example:
+
+\begin{Verbatim}
+  @HasSubsequence(value="this", from="this.start", to="this.end") int [] array;
+\end{Verbatim}
+
+The array declaration above indicates that the \code{this} object is a subsequence of \code{array},
+and that valid indexes to \code{this} range from \code{start} to \code{end}. The length of \code{this}
+is \code{end - start}. When an array is assigned to \code{array}, three facts are checked: that \code{start}
```

Comment:
Mike and I discussed it.  If the name of the subsequence is `this` then a warning isn't needed, because the definition of the sequence `this` is that it is a subsequence from start to end of the array with `@HasSubsequence`.  If more than one field is annotated with `@HasSubsequence("this", ...)` then a warning should be issued.

If the name isn't `this`, then a warning should be issued at every assignment.